### PR TITLE
feat(yum): add yum update and upgrade

### PR DESCRIPTION
**Pull request by josuemotte**
#### Pull Request (PR) description

Allow to perform in needed case `yum update` or/and `yum upgrade`

Way to use within hiera : 

```
---
yum::update: true
yum::upgrade: true
```

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -93,6 +93,15 @@
 #           baseurl: 'https://repos.example.com/CentOS/base/'
 #           mirrorlist: '--'
 #
+# @example Update:
+#   ---
+#   yum:update: true
```
Comment (TJM) [inline manifests/init.pp:98]: Hiera uses double colons between modulename and parameter

**Comment by TJM** [comment]
Documentation (yaml) bugs aside, I am not sure what you are trying to accomplish here? This would run yum upgrade (or update), which I always thought were the same operation in yum, every puppet run. I know I wouldn't want that ;) 

At a minimum, you might run it with `unless => yum check-upgrade` ... but I still wouldn't want that to run every 30 minutes.

Tommy

**Comment by josuemotte** [comment]
Hello, 

Indeed this is why I put by the default value to false, in a case of using puppet in masterless mode then it's quite useful especially for mobile workstation which are not always connected in the office. 

We thought to use something like yum-cron but it seems since those machines are not always up this solution seemed to be at least for the case of masterless the most appropriate. 

Josué

**Comment by josuemotte** [comment]
Hello @TJM , 

Do you want me to remove the `yum update` part ? 

Thank you,
Josué

**Comment by TJM** [comment]
Honestly, it is not my module @josuemotte  ;) ... so I can't make any decrees like that. I would probably stick with just upgrade *if* I wanted this feature. AS I said above, I don't want puppet running yum upgrade every 30 minutes, that seems dangerous at best. It is up to the maintainers, though ;)

**Pull request closed without merging**